infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/local/providers/LocalWorkspaceFolderPathProvider.java

```diff
@@ -44,7 +45,7 @@
     public static final String ALLOW_FOLDERS_CREATION_ENV_VARIABLE = "CHE_WORKSPACE_STORAGE_CREATE_FOLDERS";
     public static final String WORKSPACE_STORAGE_PATH_ENV_VARIABLE = "CHE_WORKSPACE_STORAGE";
 
-    private final Provider<WorkspaceManager> workspaceManager;
+    private final WorkspaceDao               workspaceDao;
```

Comment:
I do believe that provider of WorkspaceManager is better here because access to DAO should be limited